### PR TITLE
Feature/friend requests

### PR DESCRIPTION
**Pull request by PeterChu2**

### CODE REVIEW

File: app/controllers/invitations_controller.rb [hunk 1]
```diff
@@ -13,10 +13,51 @@ def new
...
+          client = SocialApiClients.getGPlus(gplus_provider)
+          plus = client.discovered_api('plusDomains')
+          userId = @invitation.friend.social_profile_auths([:gplus])[0].data.extra.raw_info.id
+          data = client.execute api_method: plus.circles.add_people,
+                                parameters: {circleId: "Following", userId: userId}
+        end
+      end
+    end
+
+    existingInvitations = Invitation.where(user_id: current_user.id, friend_id: invitation_params[:friend_id])
```
Comment (ErnestWong) [inline app/controllers/invitations_controller.rb]: we dont need this

File: app/controllers/invitations_controller.rb [hunk 2]
```diff
@@ -13,10 +13,51 @@ def new
...
+      when "gplus"
+        # TO DO: investigate why responses are 403
+        gplus_provider = current_user.authorizations.find_by_provider(:gplus)
+        if gplus_provider && friend.social_profile_linked?("gplus")
+          client = SocialApiClients.getGPlus(gplus_provider)
+          plus = client.discovered_api('plusDomains')
+          userId = @invitation.friend.social_profile_auths([:gplus])[0].data.extra.raw_info.id
+          data = client.execute api_method: plus.circles.add_people,
+                                parameters: {circleId: "Following", userId: userId}
+        end
```
Comment (ErnestWong) [inline app/controllers/invitations_controller.rb]: move somewhere else

File: app/controllers/invitations_controller.rb [hunk 3]
```diff
@@ -13,10 +13,13 @@ def new
   def create
     @invitation = current_user.invitations.build(invitation_params)
     @invitation.authorizations = get_socials
-
+    friend = @invitation.friend
+    @invitation.authorizations.each do |authorization|
+      Invitation.request(authorization.provider, current_user, friend);
+    end
```
Comment (ErnestWong) [inline app/controllers/invitations_controller.rb]: maybe we should do this in the ```@invitation.save``` block? we probaly dont want to send invitation if the ```@invitation``` is invalid

Comment (PeterChu2) [inline app/controllers/invitations_controller.rb]: fixed

File: app/controllers/invitations_controller.rb (hunk 1 above)
Comment (PeterChu2) [inline app/controllers/invitations_controller.rb]: fixed

**Pull request merged**